winch/codegen/src/isa/x64/asm.rs

```diff
@@ -688,6 +700,59 @@ impl Assembler {
         });
     }
 
+    /// Store the count of leading zeroes in src in dst.
+    /// Requires `has_lzcnt` flag.
```

Comment:
We could add an `assert!` as well in this method. Without it, the emission will fail with an error that looks like `Cannot emit inst 'lzcntl  %eax, %eax' for target; failed to match ISA requirements: [Lzcnt]` if `has_lzcnt` is false.